addons/linecompass/functions/fnc_onDraw.sqf

```diff
@@ -0,0 +1,208 @@
+
+            private _tmp = _otherMarkerControl;
+            _otherMarkerControl = _overlapCacheLineIndices param [_shiftedLineIndex, nil];
+            _overlapCacheLineIndices set [_shiftedLineIndex, _tmp];
+        };
+    } else {
+        _overlapCacheLineIndices set [_lineIndex, _control];
+    };
+
+    _nextLineMarkerControl = _nextLineMarkerControl + 1;
+
+} forEach GVAR(lineMarkers);
+
+// Remove the unused controls
+if (_nextLineMarkerControl < count GVAR(lineMarkerControlPool)) then {
+    for "_i" from _nextLineMarkerControl to (count GVAR(lineMarkerControlPool) - 1) do {
+        ctrlDelete (GVAR(lineMarkerControlPool) deleteAt _nextLineMarkerControl);
+    };
+};
+
+private _yPos = [PY(0.6), PY(2)] select GVAR(SwapOrder);
+private _hPos = PX(2.2);
+private _wPos = PY(0.3);
+
+{
+    if (ctrlShown _x) then {
+
+        private _lineIndex = _x getVariable QGVAR(lineIndex);
+        private _color = _x getVariable QGVAR(color);
+        private _alpha = (2.5 + ((_lineIndex - floor (_viewDirection / 5)) * 5) - (_viewDirection % 5)) call FUNC(getAlphaFromX);
+
+        _x ctrlSetPosition [PX(_lineIndex * 2.5 + 0.15), _yPos, _hPos, _wPos];
+        _x ctrlSetTextColor [_color select 0, _color select 1, _color select 2, (_color select 3) * _alpha];
+        _x ctrlCommit 0;
+    };
+} forEach GVAR(lineMarkerControlPool);
+
+// Icon marker
+private _nextIconMarkerControl = 0;
+
+private _player = call CBA_fnc_currentUnit;
+
+private _yOffSet = [PY(0.75), PY(2.15)] select GVAR(SwapOrder);
+
+{
+    _x params ["_unit", "_color", "_icon", "_size"];
+
+    // Check if the unit is not the player himself and alive.
+    if (alive _unit && _unit != _player && (isNull objectParent _player || {!(_unit in crew objectParent _player)})) then {
+        private _unitPosition = getPosVisual _unit;
+        private _relativeVectorToUnit = _unitPosition vectorDiff _currentPosition;
+        private _angleToUnit = ((_relativeVectorToUnit select 0) atan2 (_relativeVectorToUnit select 1) + 360) % 360;
+
+        private _control = GVAR(iconMarkerControlPool) select _nextIconMarkerControl;
+        if (isNil "_control" || {isNull _control}) then {
+            _control = _dialog ctrlCreate ["RscPicture", 7501 + _nextIconMarkerControl, _holderControl];
+            GVAR(iconMarkerControlPool) set [_nextIconMarkerControl, _control];
+        };
+
+        _control ctrlSetShadow GVAR(IconOutline);
+
+        private _compassAngle = _angleToUnit + 90;
+        if (_viewDirection >= 270 && _compassAngle < 180) then {
+            _compassAngle = _compassAngle + 360;
+        };
+        if (_viewDirection <= 90 && _compassAngle > 360) then {
+            _compassAngle = _compassAngle - 360;
+        };
+
+        _control ctrlSetText _icon;
+
+        _color set [3, ((1 - 0.2 * ((_player distance _unit) - (diwako_dui_compassRange - 6))) min 1) * ((_compassAngle - _viewDirection) call FUNC(getAlphaFromX)) min 1];
```

Comment:
oh there still is a radar dependency -> `diwako_dui_compassRange`

Either have this as a new setting or make radar a dependency again. whoops.